lib/widgets/multi_select_modal.dart

```diff
@@ -25,14 +25,23 @@ class _MultiSelectModalState extends State<MultiSelectModal> {
     );
   }
 
-  void _clearSelection() {
-    _productsStore.cleanProductAndMarketSelection();
+  void _clearSelection() async {
+    _productsStore.goToFirstPage();
+    Future.wait<dynamic>([
```

Comment:
```suggestion
    await Future.wait<dynamic>([
```

Reply:
Also, no need to pass the generic if you are not going to use the returned value.